NFT/tests/harness.rs

```diff
@@ -0,0 +1,739 @@
+
+            assert_eq!(max_supply(&owner1.nft).await, 10);
+        }
+    }
+}
+
+mod mint {
+
+    use super::*;
+
+    mod success {
+
+        use super::*;
+
+        #[tokio::test]
+        async fn mints() {
+            let (deploy_wallet, owner1, _owner2) = setup().await;
+
+            constructor(false, &deploy_wallet.nft, &Option::None(), 1).await;
+
+            let minter = Identity::Address(owner1.wallet.address());
+            assert_eq!(total_supply(&owner1.nft).await, 0);
+            assert_eq!(max_supply(&owner1.nft).await, 1);
+            assert_eq!(balance_of(&owner1.nft, &minter).await, 0);
+
+            mint(1, &owner1.nft, &minter).await;
+
+            assert_eq!(balance_of(&owner1.nft, &minter).await, 1);
+            assert_eq!(owner_of(&owner1.nft, 0).await, Option::Some(minter.clone()));
+            // Uncomment when https://github.com/FuelLabs/sway/issues/2238 is resolved
+            // assert_eq!(approved(&owner1.nft, 0).await, Option::None());
+            assert_eq!(total_supply(&owner1.nft).await, 1);
+            assert_eq!(max_supply(&owner1.nft).await, 1);
+        }
+
+        #[tokio::test]
+        async fn mints_with_access() {
+            let (deploy_wallet, owner1, _owner2) = setup().await;
+
+            let minter = Identity::Address(owner1.wallet.address());
+            let admin = Option::Some(minter.clone());
+            constructor(true, &deploy_wallet.nft, &admin, 1).await;
+
+            assert_eq!(max_supply(&owner1.nft).await, 1);
+            assert_eq!(total_supply(&owner1.nft).await, 0);
+            assert_eq!(balance_of(&owner1.nft, &minter).await, 0);
+
+            mint(1, &owner1.nft, &minter).await;
+
+            assert_eq!(balance_of(&owner1.nft, &minter).await, 1);
+            assert_eq!(owner_of(&owner1.nft, 0).await, Option::Some(minter.clone()));
+            // Uncomment when https://github.com/FuelLabs/sway/issues/2238 is resolved
+            // assert_eq!(approved(&owner1.nft, 0).await, Option::None());
+            assert_eq!(total_supply(&owner1.nft).await, 1);
+            assert_eq!(max_supply(&owner1.nft).await, 1);
+        }
+
+        #[tokio::test]
+        async fn mints_multiple() {
+            let (deploy_wallet, owner1, _owner2) = setup().await;
+
+            constructor(false, &deploy_wallet.nft, &Option::None(), 10).await;
+
+            let minter = Identity::Address(owner1.wallet.address());
+            assert_eq!(max_supply(&owner1.nft).await, 10);
+            assert_eq!(total_supply(&owner1.nft).await, 0);
+            assert_eq!(balance_of(&owner1.nft, &minter).await, 0);
+
+            mint(10, &owner1.nft, &minter).await;
+
+            assert_eq!(balance_of(&owner1.nft, &minter).await, 10);
+            assert_eq!(total_supply(&owner1.nft).await, 10);
+            assert_eq!(max_supply(&owner1.nft).await, 10);
+            for itterator in 0..10 {
+                assert_eq!(
+                    owner_of(&owner1.nft, itterator).await,
```

Comment:
```suggestion
            for iterator in 0..10 {
                assert_eq!(
                    owner_of(&owner1.nft, iterator).await,
```

is this a typo or intentional because `iterator` is a keyword?